internal/expect/client.go

```diff
@@ -0,0 +1,69 @@
+package expect
+
+import (
+	"encoding/json"
+	"errors"
+
+	opa_client "github.com/open-policy-agent/kube-mgmt/pkg/opa"
+)
+
+// Client emulates OPA Client API
+type Client struct {
+	PrefixList []string
+	// This function will be called on every request
+	actor func(req Request, value interface{}) error
+}
+
+// Prefix implements Data
+func (f *Client) Prefix(path string) opa_client.Data {
+	f.PrefixList = append(f.PrefixList, path)
+	return f
+}
+
+// PatchData implements Data.
+func (f *Client) PatchData(path string, op string, value *interface{}) (err error) {
+	req := Request{
+		req:  patchRequest,
+		path: path,
+		op:   op,
+	}
+	var actualValue interface{}
+	if value != nil {
+		actualValue = *value
+	}
+	return f.actor(req, actualValue)
+}
+
+// PutData implements Data
+func (f *Client) PutData(path string, value interface{}) (err error) {
+	req := Request{
+		req:  putRequest,
+		path: path,
+	}
+	return f.actor(req, value)
+}
+
+var errNotSupported = errors.New("PostData not supported")
```

Comment:
[nit We could inline this, it's only used once (unless I'm missing something).

Reply:
No real reason to create a variable, other than I'm used to it because it's makes it easier to check this kind of errors with `errors.Is` later. But this particular error will never be checked, so we could remove it.